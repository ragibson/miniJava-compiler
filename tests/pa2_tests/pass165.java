// PA1 parse call pass
class Test {

    void p(int a) {
	C c = new C();
	c.p(2,3);
    }
}

