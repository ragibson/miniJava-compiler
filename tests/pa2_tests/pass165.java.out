********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Test" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [1]
  .   .   . ParameterDecl
  .   .   .   INT BaseType
  .   .   .   "a"parametername 
  .   .   StmtList [2]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     ClassType
  .   .   .       "C" Identifier
  .   .   .     "c" varname
  .   .   .   NewObjectExpr
  .   .   .     ClassType
  .   .   .       "C" Identifier
  .   .   . CallStmt
  .   .   .   QualRef
  .   .   .     "p" Identifier
  .   .   .     IdRef
  .   .   .       "c" Identifier
  .   .   .   ExprList [2]
  .   .   .   . LiteralExpr
  .   .   .   .   "2" IntLiteral
  .   .   .   . LiteralExpr
  .   .   .   .   "3" IntLiteral
=============================================
